lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1210,6 +1210,40 @@ static bool ParseTBDGenArgs(TBDGenOptions &Opts, ArgList &Args,
   return false;
 }
 
+/// Determine which frame pointer setting to use. Per default if no flag is set
+/// we will use clang's default setting. This case is denoted by returning None.
+static Optional<clang::CodeGenOptions::FramePointerKind>
+determineFramePointerUse(ArgList &Args) {
+  bool requiresLeafFramePointers =
+      Args.hasArg(options::OPT_no_omit_frame_pointer_leaf);
+  bool omitAllFP = Args.hasArg(options::OPT_omit_frame_pointer);
+  bool omitLeafFP = Args.hasArg(options::OPT_omit_frame_pointer_leaf);
+  bool requiresAllFP = Args.hasArg(options::OPT_no_omit_frame_pointer);
+
+  // If the user explicitly requests -omit-frame-pointer do we emit no
+  // frame pointers.
+  if (omitAllFP)
+    return clang::CodeGenOptions::FramePointerKind::None;
+
+  // If the user explicitly requests -omit-frame-pointer do we emit no
+  // frame pointers.
+  if (omitLeafFP)
+    return clang::CodeGenOptions::FramePointerKind::NonLeaf;
```

Comment:
Assumes a default of no_omit_frame_pointer. I don't know if that's intended.

Reply:
Yes this is intended.

I don't know what case you are concerned about?

In this branch we can only have:
-fomit-frame-pointer-leaf in which case frame-pointer="non-leaf' is the right thing
-fno-omit-frame-pointer -fomit-frame-pointer-leaf in which case frame-pointer="non-leaf' is also right thing
We can't have 
-fomit-frame-pointer because that is already checked at this point.

If clang defaults to none then we will override this here.